helios-client/src/main/java/com/spotify/helios/client/HeliosClient.java

```diff
@@ -336,7 +342,32 @@ public Integer apply(final Response reply) {
   }
 
   public ListenableFuture<List<String>> listHosts() {
-    return get(uri("/hosts/"), new TypeReference<List<String>>() {
+    return listHosts(ImmutableMultimap.<String, String>of());
+  }
+
+  public ListenableFuture<List<String>> listHosts(final String namePattern) {
+    return listHosts(ImmutableMultimap.of("namePattern", namePattern));
+  }
+
+  public ListenableFuture<List<String>> listHosts(final List<String> labels) {
+    final Multimap<String, String> query = HashMultimap.create();
+    query.putAll("labels", labels);
+
+    return listHosts(query);
+  }
+
+  public ListenableFuture<List<String>> listHosts(final String namePattern,
+                                                  final List<String> labels) {
```

Comment:
The code would be less confusing to read if you used `List<HostSelector>` instead of `List<String>` here -- but I see how that would be tricky if you want to send them as query parameters (since host selectors need to encoded as json).